Add nbviewer and license badges to README

matthewfeickert opened the pull request:
The nbviewer badge added to the README is hyperlinked to the [nbviewer page for the MachineLearningStatistics repo](http://nbviewer.jupyter.org/github/dkirkby/MachineLearningStatistics/tree/master/notebooks/). This makes it easier to view the notebooks as a set of nicely rendered notes in browser (as compared to the view of the notebook that GitHub offers).

matthewfeickert committed:
Add nbviewer and license badges

The nbviewer badge is hyperlinked to the nbviewer page for the
MachineLearningStatistics repo

matthewfeickert commented:
@dkirkby I was talking with @alexarmstrongvi at CERN today and he mentioned this repo to me (quite nice). Just thought that it might be nice to add some badges to the `README` especially if people are going to try to read them as in browser static notes in addition to using the Jupyter notebooks.

Let me know if you have any requests or edits that you'd like.

dkirkby commented:
Thanks, I didn't know about nbviewer badges.  The performance (and rendering) are much nicer for read-only browsing.

Perhaps a link to the Contents notebook would be a more useful entry point than the notebooks dir?

Btw, a condensed and somewhat overlapping set of notebooks from a recent summer school are [here](https://nbviewer.jupyter.org/github/Asterics2020-Obelics/School2018/blob/master/machinelearning/Contents.ipynb).

dkirkby commented:
Are these badges documented anywhere?  I only found some [issue traffic](https://github.com/jupyter/nbviewer/issues/714).  I am wondering if there is a similar badge for starting nbviewer in the "View as slides" mode.

matthewfeickert commented:
> Perhaps a link to the Contents notebook would be a more useful entry point than the notebooks dir?

Sure. That actually works really nicely on nbviewer too.

Any interest in having a duplicate of the `Contents.ipynb` content in a `README.md` in `notebooks`? This would have the advantage that it would automatically load when the dir is visited on GitHub.

> Btw, a condensed and somewhat overlapping set of notebooks from a recent summer school are [here](https://nbviewer.jupyter.org/github/Asterics2020-Obelics/School2018/blob/master/machinelearning/Contents.ipynb).

Thanks! I'll check them out too.

matthewfeickert commented:
> Are these badges documented anywhere?

They're all generated automatically from https://shields.io/

> Is [there] a similar badge for starting nbviewer in the "View as slides" mode?

The answer seems to be kinda, but using [slideviewer](http://slideviewer.herokuapp.com/) instead of nbviewer. However, slideviewer seems to require a url to an actual Jupyter notebook, instead of just a directory above where they exist. So you could create individual badges, but maybe not just a single one. However, this was also just from 5 minutes of searching so there might be a better solution in a GitHub Issue somewhere.

For my own (local and Binder) personal presentations I just use [RISE](https://github.com/damianavila/RISE). Example [here](https://github.com/matthewfeickert/Jupyter-Notebook-Basics) (click the "launch binder" badge and then navigate to `Notebooks/Slides.ipynb` for a live demo).

matthewfeickert committed:
Link nbviewer badge to table of contents Jupyter notebook

This works out quite well, as then the user can very quickly and easily
navigate around the nbviewer page using the Contents.ipynb as a sort of
homepage

matthewfeickert commented:
@dkirkby Okay, with ad8485c1e755e5d53c73f72881053917cde21d24 the [nbviewer badge now links to the TOC notebook](http://nbviewer.jupyter.org/github/dkirkby/MachineLearningStatistics/blob/master/notebooks/Contents.ipynb).

dkirkby commented:
> Any interest in having a duplicate of the Contents.ipynb content in a README.md in notebooks?

The Contents nb already contains most of the info in the README, and I try to avoid linking to the directory.

dkirkby commented:
I just added a "slides" badge to [this README](https://github.com/Asterics2020-Obelics/School2018/blob/master/machinelearning/README.md) that links directly to nbviewer with `/format/slides/` in the URL.

matthewfeickert commented:
> The Contents nb already contains most of the info in the README, and I try to avoid linking to the directory.

Sure. Just to be clear on what I was suggesting (I may have not done a good job of describing it) was to take the TOC from `Contents.ipynb` and add that to a `notebooks/README.md` so that if someone were to visit [MachineLearningStatistics/notebooks](https://github.com/dkirkby/MachineLearningStatistics/tree/master/notebooks) they would see the rendered TOC instead of just all of the files. But maybe you're linking to the directory comment is associated with the idea that you aren't expecting people to be navigating around on GitHub so much, especially if nbviewer is in place.

> I just added a "slides" badge to [this README](https://github.com/Asterics2020-Obelics/School2018/blob/master/machinelearning/README.md) that links directly to nbviewer with `/format/slides/` in the URL.

Hey that's great! Thanks for the tip. :+1:

dkirkby commented:
> Just to be clear on what I was suggesting

Ok, I misunderstood.

> they would see the rendered TOC instead of just all of the files

Wouldn't the rendered TOC still appear below the (longish) listing of files?

> you aren't expecting people to be navigating around on GitHub so much

Yes, that's what I meant.

matthewfeickert commented:
> Wouldn't the rendered TOC still appear below the (longish) listing of files?

True. So probably not really worth it.

dkirkby commented:
Btw, are you working with Bob Kehoe at SMU?  (We collaborate on DESI).

matthewfeickert commented:
@dkirkby Nope. I work with @stephensekula on ATLAS.
